simple_backoff.go

```diff
@@ -4,22 +4,27 @@ import "time"
 
 // NewSimpleBackoff returns a simple backoff waiting the
 // specified duration longer each iteration until reset.
-func NewSimpleBackoff(step time.Duration) Backoff {
+func NewSimpleBackoff(step time.Duration, max time.Duration) Backoff {
 	return &simpleBackoff{
-		step: time.Second,
+		step: step,
+		max:  max,
 	}
 }
 
 type simpleBackoff struct {
 	current time.Duration
 	step    time.Duration
+	max     time.Duration
 }
 
 func (b *simpleBackoff) Reset() {
 	b.current = time.Duration(0)
 }
 
 func (b *simpleBackoff) Duration() time.Duration {
-	b.current += b.step
-	return b.current
+	value := b.current
+	if b.current < b.max {
+		b.current += b.step
+	}
```

Comment:
This is just a minor thing:
if `current = 100`, `step=20` and `max=110` and you run `Duration` `current` will be `120` which is larger than `max`. I think we should not exceed the max value.
```
if (b.current + b.step) <= b.max {
	b.current += b.step
}
```

Reply:
yeah true, but I thought "who cares if we'd wait a couple of seconds more. On the other hand you're right we should not exceed the limit, so let's change that.

Reply:
fixed.